sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeHealthcareEntitiesOperation.cs

```diff
@@ -116,7 +122,21 @@ public class AnalyzeHealthcareEntitiesOperation : PageableOperation<AnalyzeHealt
         /// <param name="client">The client used to check for completion.</param>
         public AnalyzeHealthcareEntitiesOperation(string operationId, TextAnalyticsClient client)
         {
-            // TODO: Add argument validation here.
+            Argument.AssertNotNullOrEmpty(operationId, nameof(operationId));
+            Argument.AssertNotNull(client, nameof(client));
+
+            try
+            {
+                OperationContinuationToken token = OperationContinuationToken.Deserialize(operationId);
+
+                _jobId = token.JobId;
+                _showStats = token.ShowStats;
+                _idToIndexMap = token.InputDocumentOrder;
+            }
+            catch (Exception e) when (e is JsonException || e is ArgumentException || e is FormatException)
```

Comment:
Given that we're only deserializing in this `try` block, should we just catch any exception that happens when deserializing (i.e.,  drop the `when`) and wrap it in this more helpful arg exception?  I think that might make this less brittle if we add any other features in `OperationContinuationToken` in the future.